File: src/main/java/com/lkimilhol/matchingproject/request/CreateMember.java
package com.lkimilhol.matchingproject.request;

import lombok.Getter;
import lombok.Setter;

import javax.validation.constraints.Min;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;

@Getter
@Setter
public class CreateMember {
    @NotNull
    @NotEmpty
    private String nickname;

    @NotNull
    @NotEmpty
    private String gender;

    @NotNull
    @Min(0)
    private int age;

    @NotNull
    @NotEmpty
    private String country;

    @NotNull
    @NotEmpty
    private String city;

    @NotNull
    @NotEmpty
    private String district;
}

File: src/main/kotlin/com/lkimilhol/matchingproject/request/CreateMember.kt
package com.lkimilhol.matchingproject.request

import javax.validation.constraints.Min
import javax.validation.constraints.NotEmpty
import javax.validation.constraints.NotNull

data class CreateMember (
    val nickname: @NotNull @NotEmpty String,
    val gender: @NotNull @NotEmpty String,
    val age : @NotNull @Min(0) Int,
    val country: @NotNull @NotEmpty String,
    val city: @NotNull @NotEmpty String,
    val district: @NotNull @NotEmpty String
)
